lib/resty/dns/resolver.lua

```diff
@@ -398,11 +414,38 @@ local function parse_response(buf, id)
         answers.errstr = resolver_errstrs[code] or "unknown"
     end
 
-    for i = 1, nan do
+    local nan_nns = nan + nns
+    local nan_nar = nan_nns + nar
+
+    local authority_section = false
+    local additional_section = false
+
+    if opts then
+        authority_section = opts.authority_section
+        additional_section = opts.additional_section
+    end
+
+    for i = 1, nan_nar do
         -- print(format("ans %d: qtype:%d qclass:%d", i, qtype, qclass))
+        local section
+        local should_skip = false
+
+        if i <= nan then
```

Comment:
@kiddkai Sure, we need to parse the 2nd section when the 3rd section is requested. But we shouldn't parse the 2nd section or the 3rd section when *neither* the 2nd *nor* the 3rd sections are requested, for example. That's my point :)